### PR TITLE
iso8601lib: add new recipe, version cci.20230123

### PR DESCRIPTION
**Pull request by vbieleny**
Specify library name and version:  **iso8601lib/cci.20230123**

I'm submitting this package because we are using this library and would like to have it as a Conan package.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/iso8601lib/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            except ConanException:
+                pass
+        try:
+            del self.settings.compiler.libcxx
+        except ConanException:
+            pass
+        try:
+            del self.settings.compiler.cppstd
+        except ConanException:
+            pass
```
Comment (StellaSmith) [inline recipes/iso8601lib/all/conanfile.py:44]:
```suggestion
        if self.options.shared:
            self.settings.rm_safe("fPIC")
         self.settings.rm_safe("compiler.cppstd")
         self.settings.rm_safe("compiler.libcxx")
```

File: recipes/iso8601lib/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    def package_info(self):
+        self.cpp_info.libs = collect_libs(self)
+
+        self.cpp_info.filenames["cmake_find_package"] = self.name
+        self.cpp_info.filenames["cmake_find_package_multi"] = self.name
+        self.cpp_info.names["cmake_find_package"] = self.name
+        self.cpp_info.names["cmake_find_package_multi"] = self.name
+
+        self.cpp_info.defines.append("ISO8601LIB_EXPORT")
```
Comment (prince-chrismc) [inline recipes/iso8601lib/all/conanfile.py:69]: This does not install the exported CMake targets https://github.com/TimSC/iso8601lib/blob/master/CMakeLists.txt to a config file so these are not needed

```suggestion
        self.cpp_info.defines.append("ISO8601LIB_EXPORT")
```

File: recipes/iso8601lib/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
+from conan import ConanFile
+from conan.errors import ConanException
```
Comment (prince-chrismc) [inline recipes/iso8601lib/all/conanfile.py:2]: Linter :)

```suggestion
```

**Review by prince-chrismc — COMMENTED** [review]
This looks super good, just some leftover targets that are not required (the default conan values are the same so less is more here)

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
c3i got stuck 😕

**Review by jwillikers — APPROVED** [review]

**Pull request merged**